compiler/damlc/tests/src/DA/Test/GenerateSimpleDalf.hs

```diff
@@ -104,6 +104,13 @@ main = do
             , tplChoices = NM.fromList ([chc,chc2] <> [arc | withArchiveChoice])
             , tplKey = Nothing
             }
+    let _syn = DefDataType -- TODO(NICK) make use of this synonym
+            { dataLocation = Nothing
+            , dataTypeCon = TypeConName ["Syn"]
+            , dataSerializable = IsSerializable True
```

Comment:
```suggestion
            , dataSerializable = IsSerializable False
```

For reference, #3635 defines synonyms, but _does not_ define them as serializable. Therefore, presence of a synonym reference in a record or variant makes that record or variant unserializable, as well. cc @remyhaemmerle-da

Reply:
Thank you. And for further reference, the LF spec for type synonyms is updated by https://github.com/digital-asset/daml/pull/3694/files

Reply:
Your are right, the spec doesn't defined synonyms as serializable. I wonder, is that right?
Why not define a type-constructor defined by a type-synonym to be serializable, so long as the synonymized type is serializable.

Reply:
I agree with @S11001001. Currently `Serializability` is defined for type not for values.  
Therefore: 
1 - we should specify how this works for type synonyms.  we need to add the following rule in the spec:
```   
     'synonym' T α₁ … αₙ ↦  σ₁  ∈  〚Ξ〛Mod  
      ⊢ₛ  σ₁[α₁ ↦ τ₁, …, αₙ ↦ τₙ]
    ———————————————————————————————————————————————————————————————— STyTypeSyn
      ⊢ₛ  Mod:T τ₁ … τₙ
```

To be consistent with the current philosophy of daml-lf, I also expect that type synonym definitions are annotated by the compiler with the `serializability` flag as any other definition.

Reply:
Sounds good :+1:

Reply:
For, say, records we require that `t_1, ..., t_n` are serializable for `Mod:T t_1 ... t_n` to be serializable. The serializability inference works under that assumption. In fact, the `serializable` flag tells whether a type preserved serializability.

Reply:
Given that we only plan to use type synonyms at the moment for typeclasses, and thanks to dictionary sanitization the underlying fields will never be serializable (they're functions from unit), I don't think we need to worry about serializability for type synonyms at the moment.

Reply:
I am fine with you proposal @associahedron. In this case, just for the sake of simplicity, set all type synonyms to Unseralizable, that is applied the change proposed by @S11001001.